### PR TITLE
Guard injection against old rubies

### PR DESCRIPTION
**Pull request by lloeki**
**What does this PR do?**

Add a known-processable frontend script to the main injection script.

**Motivation:**

Main injection script may be attempted to be loaded via `RUBYOPT=-r'/path/to/host_inject.rb'` but be unprocessable.

**Change log entry**

Add safe NOOP injection script for very old rubies

**Additional Notes:**

**How to test the change?**

### CODE REVIEW

File: lib-injection/host_inject.rb [hunk 1]
```diff
@@ -1,249 +1,7 @@
...
-      ENV['BUNDLE_GEMFILE'] = gemfile
-      utils.debug "Fork success: Using Gemfile `#{gemfile}`"
-    else
-      utils.debug 'Fork abort'
-    end
-  else
-    utils.debug 'Fork not supported... skipping injection'
-    telemetry.emit(pid, utils.version, [{ name: 'library_entrypoint.abort', tags: ['reason:fork_not_supported'] }])
-  end
+if RUBY_VERSION.start_with?('2.5.')
```
Comment (p-datadog) [inline lib-injection/host_inject.rb:5]: Should this be `>=` ?

Comment (TonyCTHsu) [reply lib-injection/host_inject.rb:5]: @p-datadog is right!

[Link to red vaccine ](https://github.com/TonyCTHsu/vaccine/actions/runs/11956259855/job/33330597924 )

**Review by TonyCTHsu — CHANGES_REQUESTED** [review]
Since the PR adds a new file, this script needs to be updated.
https://github.com/DataDog/dd-trace-rb/blob/0be95cb6681e472065ab137755982ab0283dd090/.gitlab/prepare-oci-package.sh#L7

File: lib-injection/host_inject.rb (hunk 1 above)
Comment (lloeki) [reply lib-injection/host_inject.rb]: Oh my 🤦. I guess that's what you get when writing code 5min before signing off for the day xD

File: lib-injection/host_inject.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# This file's intent is to be parseable and executable by all ruby versions
+# to call into the main one only for versions for which it is known-compatible
+# with at the language level.
+
+if RUBY_VERSION >= '2.5.'
```
Comment (lloeki) [inline lib-injection/host_inject.rb:5]: @TonyCTHsu since Vaccine is testing down to 2.3, should we lower this to 2.3?

Comment (TonyCTHsu) [reply lib-injection/host_inject.rb:5]: We can do that, but vaccine is testing the last patch version of 2.3

```
% docker run --rm ghcr.io/datadog/images-rb/engines/ruby:2.3 bash -c 'ruby -e "puts RUBY_VERSION"'
2.3.8
```

**Review by TonyCTHsu — APPROVED** [review]

File: lib-injection/host_inject.rb (hunk 2 above)
Comment (lloeki) [reply lib-injection/host_inject.rb:5]: Technically true, although as far as parsing/processing goes 2.3.0 to 2.3.8 are the same, so I expect it to be OK.

**Pull request merged**